web/client/actions/wfsquery.js

```diff
@@ -139,12 +141,13 @@ function createQuery(searchUrl, filterObj) {
     };
 }
 
-function query(searchUrl, filterObj, queryOptions) {
+function query(searchUrl, filterObj, queryOptions, reason) {
```

Comment:
I understand why you need this additional field. Action sequence is not enough to understand if activate or not autoselect? like geometryFilterActive + updateFilter? It should work when you select from the feature grid and when you start from the Feature Info
If it is not, please document well this method and the point you use to intercept this.